### PR TITLE
core: Correctly format nested outputs

### LINKED ISSUE #7143: Unable to make splat reference to attribute of type map
### Terraform Version

master / 2127466280172a7cd0aa1d863cb932aa0107c986 

### Affected Resource(s)

core

### Terraform Configuration Files
```hcl
resource "aws_security_group" "foo" {
  count = 3
  tags {
    Name = "sg-${count.index}"
  }
}

resource "aws_instance" "web" {
  count                  = 2
  instance_type          = "t2.nano"
  ami                    = "ami-06b94666"
  vpc_security_group_ids = ["${aws_security_group.foo.*.id}"]
}

output "listoflists" { value = ["${aws_instance.web.*.vpc_security_group_ids}"] }
output "listofmaps" { value = ["${aws_security_group.foo.*.tags}"] }
```

### Steps to Reproduce

1. `TF_LOG=warn terraform apply`

### Expected Behavior

Outputs including lists of lists and list of maps.

### Actual Behavior

The `listoflists` output is flattened (perhaps this is intentional behavior?):

```
listoflists = [
  sg-553ce633
  sg-543ce632
  sg-523ce634
  sg-553ce633
  sg-543ce632
  sg-523ce634
]
```

And the `listofmaps` output cannot be referenced. Two warnings are printer:

```
2016/06/12 12:17:01 [WARN] Output interpolation "listofmaps" failed: Resource 'aws_security_group.foo' does not have attribute 'tags' for variable 'aws_security_group.foo.*.tags'
2016/06/12 12:17:01 [WARN] Output interpolation "listofmaps" failed: Resource 'aws_security_group.foo' does not have attribute 'tags' for variable 'aws_security_group.foo.*.tags'
```

### PR DESCRIPTION
**Pull request by jen20**
This commit pretty prints outputs which consist of nested complex structures (e.g. lists of lists, lists of maps).

Fixes #7143.

cc @phinze

### CODE REVIEW

**Comment by jen20** [comment]
Formating fixed now @phizne!

**Comment by phinze** [comment]
I would love to see a unit test or two around this, but since it's cosmetic and I've seen it demo'd I'll allow it! 😄 👍 LGTM

**Comment by jen20** [comment]
@phinze I've added some tests and a new `-json` flag to the output format - detail in the commit message. I'll update the docs and push to this branch.

File: command/output_test.go [hunk 1]
```diff
@@ -53,19 +53,19 @@ func TestOutput(t *testing.T) {
 func TestModuleOutput(t *testing.T) {
 	originalState := &terraform.State{
 		Modules: []*terraform.ModuleState{
-			&terraform.ModuleState{
+			{
```
Comment (phinze) [inline command/output_test.go]: 😄

**Comment by phinze** [comment]
LGTM!

**Pull request merged**